cmd/kube-controller-manager/app/core.go

```diff
@@ -92,7 +92,8 @@ func newServiceLBControllerDescriptor() *ControllerDescriptor {
 }
 
 func startServiceLBController(ctx context.Context, controllerContext ControllerContext, controllerName string) (controller.Interface, bool, error) {
-	serviceController, err := servicecontroller.New(
+	serviceController, err := servicecontroller.NewWithContext(
```

Comment:
side note, @elmiko @thockin @liggitt @dims I wonder if we should still be running this controllers in the kube-controller-manager or completely remove them ... will they even work after the removal of the cloud provider code?

Reply:
more on https://github.com/kubernetes/kubernetes/issues/128121#issuecomment-2422075168

Reply:
/hold

@aojea @pohly I find another way to fix this issue and will create a new PR. It is an old bug in the service controller. I am testing it with my fix.

Reply:
so, cloud-provider name must no be empty https://github.com/kubernetes/kubernetes/blob/85ff7e75a2066fef3543a55ca995c5173e779b72/staging/src/k8s.io/cloud-provider/options/options.go#L271-L273

Reply:
@aojea @pohly  Another PR to fix an old issue: https://github.com/kubernetes/kubernetes/pull/128182